pkg/apis/policy/v1alpha1/propagation_types.go

```diff
@@ -42,6 +42,9 @@ type PropagationPolicy struct {
 // PropagationSpec represents the desired behavior of PropagationPolicy.
 type PropagationSpec struct {
 	// ResourceSelectors used to select resources.
+	// Nil selector is not allowed and an empty selector selects nothing for
```

Comment:
As far as I know, `resourceSelectors` are not allowed to be modified, so it is meaningless for users to create empty resource selectors. If users are not prevented from creating empty resource selectors, will the empty resource selectors accidentally created by users become technical debt?

Reply:
Good questions.

> As far as I know, resourceSelectors are not allowed to be modified,

We are going to enable it. See #2562.

> so it is meaningless for users to create empty resource selectors.

Yeah, I agree with that. a PropagationPolicy with an empty `ResourceSelector` looks weird. 
I don't know if the `kubebuilder:` markers can help to restrict the length.(restrict to len(resourceSelectors) > 0). If not, we can add the restriction logic to `karmada-webhook`. How do you think?

Reply:
Just checked the marker, maybe `+kubebuilder:validation:MinItems:=1` would help.
https://book.kubebuilder.io/reference/markers/crd-validation.html

Reply:
> +kubebuilder:validation:MinItems:=1

+1

Reply:
Done. and I put the test report into to PR description.